core/field_angle.ts

```diff
@@ -133,17 +133,16 @@ export class FieldAngle extends FieldInput<number> {
    * for a list of properties this parameter supports.
    */
   constructor(
-      opt_value?: string|number|Sentinel, opt_validator?: FieldAngleValidator,
+      opt_value?: number|string|Sentinel, opt_validator?: FieldAngleValidator,
       opt_config?: FieldAngleConfig) {
     super(Field.SKIP_SETUP);
 
-    if (opt_value === Field.SKIP_SETUP) {
-      return;
-    }
+    if (Field.isSentinel(opt_value)) return;
     if (opt_config) {
       this.configure_(opt_config);
     }
-    this.setValue(opt_value);
+    const value = opt_value === undefined ? null : opt_value;
```

Comment:
in the old code:

1. constructor calls super which sets value to `DEFAULT_VALUE` which is `0`
1.  `setValue_(undefined)`
2. calls `doClassValidation_(undefined)`
3. returns `null` (because like you said `Number(undefined) === null`)
4. `processValidation_(null)` means we don't actually change the value. we call `doValueInvalid_` but the actual value is never changed, thus it stays as `DEFAULT_VALUE`

So the old code is actually fine. But even if it weren't, I'd say the important thing is keeping the behavior the same as it used to be. So that means letting the class Validator run if it used to.

Reply:
Ope - looks like I overlooked `processValidation_` then. Glad to see the old implementation is fine!

Sounds good - I'll update `setValue` to accept undefined and revert the old behavior.